nep-9.mediawiki

```diff
@@ -0,0 +1,189 @@
+
+==Motivation==
+
+Currently, there is no standard URI for NEO clients to consume. Bitcoin implements a URI standard so that clicking links or scanning QR codes can easily implement a bitcoin payment. Similarly the transfer of native assets on the NEO chain should be jat as simple. 
+
+However NEO can possibly implement much more contract operations via smart contract execution. These can range from minting and transferring tokens, registering domain names, exchanges, and other customized applications. Ideally a URI could exist for generic smart contract execution. However, this has security implications, as it may be difficult to determine what a smart contract URI actually does, which could lead to loss of funds.
+
+To counter this we propose that the URI's generated for smart contract invocation to be limited to a subset of well established contract operations like NEP-5 token transfer. This will allow a client to easily understand and verify what a given URI will be to do. If the client is even more security conscious it could maintain a white list of contracts that follow the given URI spec.
+
+Any new NEO URI that is added to this specification should be used for a smart contract that is secure, well documented, and commonly used. We will discuss more about this at the end of this proposal.
+
+==Specification==
+
+===Native Asset Transfer URI===
+
+A native asset transfer would have the following URI. It describes the receiver address, assetID, and additional attributes that will be sent with the transaction. 
+
+'''URI scheme: <code>neo</code>'''
+
+<code>neo:<address>[?assetID=<assetID>][?amount=<amount>][?<TransactionAttributeKey>=<value>]</code>
+
+==== URI Keys====
+{| class="wikitable"
+! Key
+! Description
+! Required
+|-
+| address
+| Valid NEO Address
+| ✓
+|-
+| assetID
+| ID of an asset being sent.
+| -
+|-
+| amount
+| amount of asset being sent. e.g 1.0
+| -
+|}
+
+==== Available URI keys map to NEO transaction attribute keys ====
+{| class="wikitable"
+! URI Key
+! NEO Transaction attribute
+! Description
+|-
+| contractHash
+| 0x00
+| Hash value of contract 
+|-
+| ecdh02, ecdh03
+|	0x02,0x03
+|	Public key for ECDH key exchange
+|-
+| script 
+|	0x20
+|	Additional validation of transactions
+|-
+| vote
+|	0x30
+| For voting
+|-
+| certUrl
+| 0x80 
+|	Url address of certificate 
+|-
+| descriptionUrl
+| 0x81
+|	Url address of description 
+|-
+| description
+| 0x90
+| Brief description 
+|-
+| hash1,hash2,...,hash15
+|	0xa1,0xa2,...,0xaf
+|	Used to store custom hash values
+|-
+| remark,remark2,...,remark15
+|	0xf0,0xf1,...,0xff
+|	Remarks
+|}
+
+====Native Assets====  
+{| class="wikitable"
+! Name
+! Hash
+|-
+| NEO 
+| c56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b 
+|-
+| GAS 
+| 602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7 
+|}
+
+====Examples====
+
+Begin transaction to specified address
+
+<code>neo:AeNkbJdiMx49kBStQdDih7BzfDwyTNVRfb</code>
```

Comment:
Are URIs case sensitive? It would be troubling if the case is ignored for addresses.